src/database/migrations/0069-create-files.sql

```diff
@@ -0,0 +1,10 @@
+CREATE TABLE files (
+	uuid uuid PRIMARY KEY DEFAULT gen_random_uuid(),
```

Comment:
Rather than identifying files by a random number, files self-identify by checksum. We could use `bytea` of some length (256 bits?) and use the `b2sum` of the file to identify it. The only trouble with my suggestion is that then it naturally de-duplicates these file records if the checksum is the primary key. So the key would have to be across multiple fields.

Reply:
> Reflecting on all of the above: I'm thinking we should call this `storage_key` instead of `uuid` which makes it more explicit that this field is ultimately the key associated with the file in the storage service (or `s3_key` but that feels too implementation-specific).
> 
> I think with that name it's more intuitive as to why we would want that to be a generated UUID rather than a derived checksum

The S3 API is exposed to the client in this flow. So 'tis not a problem to make it explicit.

Regardless, I'd be fine with `storage_key`.

Reply:
I meant less safe in terms of "Anybody can predict a file's location on the internet based on its contents"

I do agree that storing a checksum and using that checksum is a good idea for a future augmentation of our presigned post flow; I just don't think it should be the identifier.

Reply:
Created https://github.com/PhilanthropyDataCommons/service/issues/1871 (resolving this thread since it's now over there)